### PR TITLE
Support migration and publication of course verification deadlines

### PR DESCRIPTION
**Pull request by rlucioni**
XCOM-537.

@clintonb @jimabramson

### CODE REVIEW

**Comment by clintonb** [comment]
`CourseSerializer` needs to be updated.

**Comment by rlucioni** [comment]
@clintonb updated.

**Comment by clintonb** [comment]
:+1:

File: ecommerce/extensions/catalogue/management/commands/migrate_course.py [hunk 1]
```diff
@@ -35,40 +38,65 @@ def _build_lms_url(self, path):
...
+            ))
 
         data = response.json()
         logger.debug(data)
+
         course_name = data['name']
+        if course_name is None:
+            message = u'Aborting migration. No name is available for {}.'.format(self.course.id)
+            logger.error(message)
+            raise Exception(message)
```
Comment (jimabramson) [inline ecommerce/extensions/catalogue/management/commands/migrate_course.py:65]: question: if migrating a bunch of courses, and this happens somewhere midway through, will all the successfully migrated courses up til that point still be committed in Otto?

**Comment by jimabramson** [comment]
:+1: other than my question about transaction scope / exception handling

File: ecommerce/extensions/catalogue/management/commands/migrate_course.py (hunk 1 above)
Comment (rlucioni) [inline ecommerce/extensions/catalogue/management/commands/migrate_course.py:65]: My understanding is that yes, those successfully migrated courses will be committed. We're using `transaction.atomic()` to explicitly delimit a block of operations that will be rolled back in the event of an exception. That block of operations is attempted repeatedly in the case of a bulk migration, once for each course; successful runs are committed, while those which encounter exceptions are not.

**Pull request merged**